root_draft_metadata.tf

```diff
@@ -46,9 +47,212 @@ module "draft_metadata_api_gateway" {
+  name               = "TDRConsignmentAPIConnection${title(local.environment)}"
+  authorization_type = "OAUTH_CLIENT_CREDENTIALS"
+
+  auth_parameters {
+    oauth {
+      client_parameters {
+        client_id     = local.keycloak_backend-checks_client_id
+        client_secret = data.aws_ssm_parameter.backend_checks_keycloak_secret.value
+      }
+
+      authorization_endpoint = "${local.keycloak_auth_url}/realms/tdr/protocol/openid-connect/token"
+      http_method            = "POST"
+
+      oauth_http_parameters {
+        body {
+          key             = "grant_type"
+          value           = "client_credentials"
+          is_value_secret = false
+        }
+      }
+    }
+  }
+}
+
+resource "aws_iam_policy" "draft_metadata_checks_policy" {
+  name        = "TDRMetadataChecksPolicy${title(local.environment)}"
+  description = "Policy to allow necessary lambda executions from step function"
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Action = [
+          "lambda:InvokeFunction"
+        ]
+        Resource = [
+          module.yara_av_v2.lambda_arn,
+          module.draft_metadata_validator_lambda.lambda_arn
+        ]
+      }
+    ]
+  })
+}
+
+resource "aws_iam_policy" "api_invoke_policy" {
+  name = "TDRAPIInvokePolicy${title(local.environment)}"
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect   = "Allow"
+        Action   = "states:InvokeHTTPEndpoint"
+        Resource = module.draft_metadata_checks.step_function_arn
+        Condition = {
+          StringEquals = {
+            "states:HTTPMethod" = "POST"
+          }
+          StringLike = {
+            "states:HTTPEndpoint" = "${module.consignment_api.api_url}/*"
+          }
+        }
+      },
+      {
+        Effect   = "Allow"
+        Action   = "events:RetrieveConnectionCredentials"
+        Resource = aws_cloudwatch_event_connection.consignment_api_connection.arn
+      },
+      {
+        Effect = "Allow"
+        Action = [
+          "secretsmanager:GetSecretValue",
+          "secretsmanager:DescribeSecret"
+        ]
+        Resource = "arn:aws:secretsmanager:${local.region}:${var.tdr_account_number}:secret:events!connection/*"
+      }
+    ]
+  })
+}
+
+module "draft_metadata_checks" {
+  source             = "./da-terraform-modules/sfn"
+  step_function_name = "TDRMetadataChecks${title(local.environment)}"
+  step_function_definition = jsonencode(
+    {
+      "Comment" : "Run antivirus checks on metadata, update DB if positive, else trigger metadata validation",
+      "StartAt" : "RunAntivirusLambda",
+      "States" : {
+        "RunAntivirusLambda" : {
+          "Type" : "Task",
+          "Resource" : module.yara_av_v2.lambda_arn
+          "Parameters" : {
+            "consignmentId.$" : "$.consignmentId",
+            "fileId" : "draft-metadata.csv",
```

Comment:
This probably should not be hard coded, in case the name is changed for the CSV file etc. Can it be passed in from the step function request?